app/models/match_decisions/match_recommendation_shelter_agency.rb

```diff
@@ -132,9 +132,9 @@ def expiration_update
 
       end
 
-      def canceled
+      def canceled contact
         Notifications::MatchCanceled.create_for_match! match
-        match.canceled!
+        match.canceled! contact&.id
```

Comment:
`ClientOpportunityMatch#canceled!` takes a named parameter, so this won't work as written.